content/general/concepts/what-is-the-factory-pattern/what-is-the-factory-pattern.md

```diff
@@ -0,0 +1,273 @@
+---
+Title: 'What is the Factory Pattern?'
+Description: 'The factory pattern defers instantiation logic of a parent abstract class to its concrete sub-classes.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The _factory pattern_ defers instantiation logic of a parent abstract class to its concrete sub-classes. At time of object creation, the specific class type may not be known, in which a creator class' factory method is used to decouple this identifying logic. This is typically done through the methods parameters and a switch statement.
+
+## UML Design
+
+![UML diagram of a factory](/media/factory-uml.png)
```

Comment:
```suggestion
![UML diagram of a factory](https://raw.githubusercontent.com/Codecademy/docs/main/media/factory-uml.png)
```

Reply:
It might be an idea to put this URL in a readme. 

Apologies if I missed it